evennia/commands/default/building.py

```diff
@@ -2778,8 +2778,35 @@ def func(self):
             switches.append("loc")
 
         searchstring = self.lhs
-        low, high = 1, ObjectDB.objects.all().order_by("-id")[0].id
+        
+        try:
+            # Try grabbing the actual min/max id values by database aggregation
+            qs = ObjectDB.objects.values('id').aggregate(low=Min('id'), high=Max('id'))
+            low, high = sorted(qs.values())
+            if not (low and high):
+                raise ValueError(f"{self.__class__.__name__}: Min and max ID not returned by aggregation; falling back to queryset slicing.")
+        except Exception as e:
+            logger.log_trace(e)
+            # If that doesn't work for some reason (empty DB?), guess the lower 
+            # bound and do a less-efficient query to find the upper.
+            low, high = 1, ObjectDB.objects.all().order_by("-id")[0].id
+            
         if self.rhs:
+            # Check that rhs is either a valid dbref or dbref range
+            try:
+                # Get rid of # signs, split on hyphen or space and cast all to int.
+                # Then sort by number to get the lowest and highest values
+                # comprising the bounds.
+                bounds = sorted(int(x) for x in re.split('[-\s]+', self.rhs.strip().replace('#', '')))
```

Comment:
I see that this command uses raw stripping of `#` in other places as well, but these days it may be a good idea to centralize parsing like this by using the special helper `evennia.utils.utils.dbref()` which takes a string and checks if it's a valid `#dbref` structure before spitting out the raw `id`. It returns `None` if the dbref structure is invalid (such as `#foo`).

Reply:
Easy enough to fix.